packages/flutter/lib/src/widgets/focus_traversal.dart

```diff
@@ -682,160 +948,607 @@ class _SortData {
+
+    // Pick the initial widget as the one that is at the beginning of the band
+    // of the topmost, or the topmost, if there are no others in its band.
+    _ReadingOrderSortData current = _pickNext(unplaced);
+    sortedList.add(current.node);
     unplaced.remove(current);
 
+    // Go through each node, picking the next one after eliminating the previous
+    // one, since removing the previously picked node will expose a new band in
+    // which to choose candidates.
     while (unplaced.isNotEmpty) {
-      final _SortData next = pickFirst(unplaced);
+      final _ReadingOrderSortData next = _pickNext(unplaced);
       current = next;
-      sortedList.add(current);
+      sortedList.add(current.node);
       unplaced.remove(current);
     }
-    return sortedList.map((_SortData item) => item.node);
+    return sortedList;
   }
+}
 
-  // Moves the focus forward or backward in reading order, depending on the
-  // value of the forward argument.
-  bool _move(FocusNode currentNode, {@required bool forward}) {
-    final FocusScopeNode nearestScope = currentNode.nearestScope;
-    invalidateScopeData(nearestScope);
-    final FocusNode focusedChild = nearestScope.focusedChild;
-    if (focusedChild == null) {
-      final FocusNode firstFocus = findFirstFocus(currentNode);
-      if (firstFocus != null) {
-        _focusAndEnsureVisible(
-          firstFocus,
-          alignmentPolicy: forward
-              ? ScrollPositionAlignmentPolicy.keepVisibleAtEnd
-              : ScrollPositionAlignmentPolicy.keepVisibleAtStart,
-        );
-        return true;
+/// Base class for all sort orders for [OrderedTraversalPolicy] traversal.
+///
+/// {@template flutter.widgets.focusorder.comparable}
+/// Only orders of the same type are comparable. If a set of widgets in the same
+/// [FocusTraversalGroup] contains orders that are not comparable with each other, it
+/// will assert, since the ordering between such keys is undefined. To avoid
+/// collisions, use a [FocusTraversalGroup] to group similarly ordered widgets
+/// together.
+/// {@endtemplate}
+///
+/// See also:
+///
+///  * [FocusTraversalGroup], a widget that groups together and imposes a
+///    traversal policy on the [Focus] nodes below it in the widget hierarchy.
+///  * [FocusTraversalOrder], a widget that assigns an order to a widget subtree
+///    for the [OrderedFocusTraversalPolicy] to use.
+///  * [NumericFocusOrder], for a focus order that describes its order with a
+///    `double`.
+///  * [LexicalFocusOrder], a focus order that assigns a string-based lexical
+///    traversal order to a [FocusTraversalOrder] widget.
+@immutable
+abstract class FocusOrder extends Diagnosticable implements Comparable<FocusOrder> {
+  /// Abstract const constructor. This constructor enables subclasses to provide
+  /// const constructors so that they can be used in const expressions.
+  const FocusOrder();
+
+  @override
+  @nonVirtual
+  int compareTo(FocusOrder other) {
```

Comment:
The class doc should probably mention that you should override doCompare instead of compareTo.